src/conduits/RwaOutputConduit3.sol

```diff
@@ -0,0 +1,261 @@
+    mapping(address => uint256) public wards;
+    /// @notice Addresses with operator access on this contract. `can[usr]`
+    mapping(address => uint256) public can;
+
+    /// @dev This is declared here so the storage layout lines up with RwaOutputConduit.
+    DSTokenAbstract private __unused_gov;
+    /// @notice Dai token contract address
+    DSTokenAbstract public dai;
+    /// @notice Dai output address
+    address public to;
+
+    /// @dev This is declared here so the storage layout lines up with RwaOutputConduit.
+    mapping(address => uint256) private __unused_bud;
+    /// @notice Addresses with push access on this contract. `may[usr]`
+    mapping(address => uint256) public may;
+
+    /// @notice Exit address
+    address public quitTo;
+
+    /**
+     * @notice `usr` was granted admin access.
+     * @param usr The user address.
+     */
+    event Rely(address indexed usr);
+    /**
+     * @notice `usr` admin access was revoked.
+     * @param usr The user address.
+     */
+    event Deny(address indexed usr);
+    /**
+     * @notice `usr` was granted push access.
+     * @param usr The user address.
+     */
+    event Mate(address indexed usr);
+    /**
+     * @notice `usr` push access was revoked.
+     * @param usr The user address.
+     */
+    event Hate(address indexed usr);
+    /**
+     * @notice `usr` was granted operator access.
+     * @param usr The user address.
+     */
+    event Hope(address indexed usr);
+    /**
+     * @notice `usr` operator access was revoked.
+     * @param usr The user address.
+     */
+    event Nope(address indexed usr);
+    /**
+     * @notice `wad` amount of Dai was pushed to the recipient `to`.
+     * @param to The Dai recipient address
+     * @param wad The amount of Dai
+     */
+    event Push(address indexed to, uint256 wad);
+    /**
+     * @notice A contract parameter was updated.
+     * @param what The changed parameter name. Currently the supported values are: "quitTo", "to".
+     * @param data The new value of the parameter.
+     */
+    event File(bytes32 indexed what, address data);
+    /**
+     * @notice The conduit outstanding gem balance was flushed out to `quitTo` address.
+     * @param quitTo The quitTo address.
+     * @param wad The amount flushed out.
+     */
+    event Quit(address indexed quitTo, uint256 wad);
+
+    /**
+     * @notice Defines PSM and quitTo addresses and gives `msg.sender` admin access.
+     * @param _psm PSM contract address.
+     * @param _quitTo Address to where outstanding GEM balance will go after `quit`
+     */
+    constructor(address _psm, address _quitTo) public {
+        DSTokenAbstract _gem = DSTokenAbstract(GemJoinAbstract(PsmAbstract(_psm).gemJoin()).gem());
+        psm = PsmAbstract(_psm);
+        dai = DSTokenAbstract(PsmAbstract(_psm).dai());
+        gem = _gem;
+        quitTo = _quitTo;
+        toGemConversionFactor = 10**(dai.decimals() - _gem.decimals());
```

Comment:
We should probably safe-guard against `_gem` having more than 18 decimals:

```suggestion
        uint256 gemDecimals = gem.decimals();
        require(gemDecimals <= 18, "RwaOutputConduit3/invalid-gem-decimals");
        toGemConversionFactor = 10**(dai.decimals() - gemDecimals);
```

Reply:
yes, already working on it

Reply:
We cannot guarantee all existing PSMs now and in the future will use `AuthGemJoin5`, can we?
This is something that can easily be overlooked.

Since this is done in the constructor only, I don't think an extra require is that much of a problem.

Reply:
Done